orb_models/forcefield/pretrained.py

```diff
@@ -39,17 +39,21 @@ def get_base(
 
 
 def load_model_for_inference(
-    model: torch.nn.Module, 
-    weights_path: str, 
+    model: torch.nn.Module,
+    weights_path: str,
     device: Union[torch.device, str] = None,
 ) -> torch.nn.Module:
     """Load a pretrained model in inference mode, using GPU if available."""
     local_path = cached_path(weights_path)
     state_dict = torch.load(local_path, map_location="cpu")
-
     model.load_state_dict(state_dict, strict=True)
-    model = model.to(get_device(device))
-
+    device = get_device(device)
+    if device.type == "cuda" and torch.get_float32_matmul_precision() == "high":
```

Comment:
This will warn every time a model is loaded. Instead, we can add a global variable and check if we have warned before, so the warning only ever appears once.